cluster/external/provider.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+function _kubectl() {
+    kubectl "$@"
+}
+
+# The external cluster is assumed to be up.  Do a simple check
+function up() {
+    _kubectl version >/dev/null
+    if [ $? -ne 0 ]; then
+        echo -e "\n*** Unable to reach external cluster.  Please check configuration ***"
+        echo -e "*** Type \"kubectl config view\" for current settings               ***\n"
+        exit 1
+    fi
+    echo "Cluster is up"
+}
+
+function down() {
+    echo "Not supported by this provider"
+}
+
+function build() {
+    echo "Not supported by this provider"
```

Comment:
To understand the context on how you want to use the provider: Is your intent to do the whole `make cluster-sync` stuff which builds the code, removes old versions from the cluster, pushes new images to the registry and deploys the new manifests, externally?

In the other providers we set a proper `docker_prefix` in the config which points to the used docker registry and let `make cluster-sync` do all required updates.

Reply:
@rmohr: here's how I currently do things:
make build --> compile stuff
make push --> push images to my Docker registry (based on DOCKER_PREFIX and DOCKER_TAG)
make cluster-sync --> resolve kubevirt installation (remove existing manifests, add new ones)

We're currently using make as a really crappy shell script language, and having make call scripts that in turn call make is less than ideal.  If we can improve the performance of the dockerized operations, we can use make properly with dependencies, and then maybe 'cluster sync' or some other wrapper target can do everything in one shot.

Until then, this works, and the workflow is fairly straightforward.

Reply:
> We're currently using make as a really crappy shell script language, and having make call scripts that in turn call make is less than ideal.

@benwatcisco oh, we use the Makefile extra only as well-known entrypoints, not in the sense of make. Further the functions for providers are extra not called in the dockerized environment so that you can for instance also use the Makefile targets there. For as long as you don't call `cluster-sync` from in there again, you should be able to do there what you want.

Reply:
So taking the flow which you described here:

> make build --> compile stuff
> make push --> push images to my Docker registry (based on DOCKER_PREFIX and DOCKER_TAG)
> make cluster-sync --> resolve kubevirt installation (remove existing manifests, add new ones)

You could make your `build` function look like this:

```bash
function build() {
 # does "build" in docker, create from the result outside of docker, pushes to the registry specified with DOCKER_PREFIX and DOCKER_TAG
make push
# creates manifests under consideration of DOCKER_PREFIX and DOCKER_TAG
make manifests
}
```

You could still save one extra docker call by somehow merging the `build` and `manifests` phase. In my case that would only save about two seconds.

But maybe I silll miss something.

Reply:
@benwatcisco The workflow you describe is nice and simple, I like it.  I do find that the point @rmohr makes here about just wrapping the steps up in a ```make build``` extremely nice, and more importantly consistent with the other providers which is a bug plus.  Are you still interested in this effort?  I'm happy to pick it up if you don't have the time, or to help test/review it if you're still into it.  Thanks!!

Reply:
@j-griffith I would like to see this code pulled in to master, and am willing to push it through.  I personally don't like the idea of calling make this way, and will try to come up with something better, but now now :)